tests/rptest/tests/upgrade_test.py

```diff
@@ -128,3 +129,47 @@ def test_rolling_upgrade(self):
         post_upgrade_num_msgs = self.producer.num_acked
         self.run_validation(min_records=post_upgrade_num_msgs +
                             (self.producer_msgs_per_sec * 3))
+
+    @cluster(num_nodes=5, log_allow_list=RESTART_LOG_ALLOW_LIST)
+    @parametrize(upgrade_after_rollback=True)
+    @parametrize(upgrade_after_rollback=False)
+    def test_rolling_upgrade_with_rollback(self, upgrade_after_rollback):
+        self.installer.install(self.redpanda.nodes, RedpandaInstaller.HEAD)
+
+        # Upgrade one node.
+        first_node = self.redpanda.nodes[0]
+        # Give ample time to restart, given the running workload.
+        self.redpanda.restart_nodes([first_node],
+                                    rolling_restart=True,
+                                    start_timeout=90,
+                                    stop_timeout=90)
+
+        def await_progress():
+            num_msgs = self.producer.num_acked
+            self.await_num_produced(num_msgs +
+                                    (self.producer_msgs_per_sec * 3))
```

Comment:
What is the reason for multiply by 3 instead of another number?

Reply:
It's just a golidlocks number, I don't want it to be really high so the test isn't slow, but it's high enough we should be confident multiple messages are making it through